File: leetcode3/남효정/538. Convert BST to Greater Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.total = 0

        def reversed(node):
            if not node:
                return 
            
            reversed(node.right)

            self.total += node.val
            node.val = self.total

            reversed(node.left)
        
        reversed(root)
        return root

File: leetcode3/황은지/538. Convert BST to Greater Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function (root) {
  function changeValue(preSum, root) {
    if ((root === null) | (root === undefined)) return 0;
    changeValue(0, root.right);
    const sumRight = root.right ? root.right.val : 0;
    root.val = root.val + preSum + sumRight;
    changeValue(root.val, root.left);
  }

  changeValue(0, root);
  return root;
};
